### DISCUSSION TITLE
Select with NativeSelect style keyboard selection

### DISCUSSION THREAD

**Discussion by digitaltoad** [mantinedev/mantine#2272] [Q&A]
Is there currently any way to get `Select` to automatically highlight the best match from keyboard filtering? A native select will filter and select a match from a few keystrokes that match a label. Adding `searchable` to `Select` only partially helps here because none of the options are actually highlighted while typing which means `selectOnBlur` doesn't work either. I know I can use `NativeSelect` if I want this style of form navigation, but not if I want a custom dropdown. Any help would be greatly appreciated.

**Comment by rtivital** [comment]
Hi @digitaltoad, none of the items get hovered state when user searches because we got reported multiple times that it is unwanted behavior – you can select items with arrows.

**Reply by digitaltoad** [reply]
So because this behavior is closer to default behavior of a native select, could this functionality be a flag on the select? At least if it's configurable, if someone doesn't want it, they don't have to use it. I wouldn't mind submitting a PR for it.

**Reply by rtivital** [reply]
Sure, feel free to send a PR
